### PR TITLE
Histogram

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Histogram](https://trello.com/c/IB3g5DCS/329-histogram)

**Comment by michaelmattig** [comment]
grafik

- I don't see any labels yet, is that intended?
- the bars are not aligned with the ticks

**Comment by michaelmattig** [comment]
when going into full screen the bars don't get wider, even though there is space

grafik
grafik

File: projects/wave-core-new/src/lib/backend/backend.model.ts [hunk 1]
```diff
@@ -191,15 +193,52 @@ export interface InternalDataSetIdDict {
...
     [key: string]: number;
 }
+
+export interface PlotDataDict {
+    plot_name: string;
+    plot_type: 'Json' | 'Chart' | 'Png';
+    data: any;
+}
+
+export interface WorkflowMetadataDict {
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/backend/backend.model.ts:215]: isn't the workflowmetadata just the ResultDescriptor?

File: projects/wave-core-new/src/lib/backend/backend.model.ts [hunk 2]
```diff
@@ -191,15 +193,52 @@ export interface InternalDataSetIdDict {
...
+    spatial_reference?: SrsString;
+}
+
+export interface RasterMetadataDict extends WorkflowMetadataDict {
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/backend/backend.model.ts:219]: redundant to RasterResultDescriptorDict?

File: projects/wave-core-new/src/lib/backend/backend.model.ts [hunk 3]
```diff
@@ -191,15 +193,52 @@ export interface InternalDataSetIdDict {
...
+    data_type: 'U8' | 'U16' | 'U32' | 'U64' | 'I8' | 'I16' | 'I32' | 'I64' | 'F32' | 'F64';
+    measurement: 'unitless' | MeasurementDict;
+}
+
+export interface VectorMetadataDict extends WorkflowMetadataDict {
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/backend/backend.model.ts:224]: redundant to ResultDescriptorDict?

File: projects/wave-core-new/src/lib/backend/backend.service.ts [hunk 4]
```diff
@@ -150,6 +153,12 @@ export class BackendService {
         });
     }
 
+    getWorkflowMetadata(workflowId: UUID, sessionId: UUID): Observable<RasterMetadataDict | VectorMetadataDict | PlotMetadataDict> {
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/backend/backend.service.ts:156]: should return ResultDescriptor

File: projects/wave-core-new/src/lib/layers/layer-metadata.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import {HasLayerType, LayerType} from './layer.model';
+import {RasterMetadataDict, VectorMetadataDict} from '../backend/backend.model';
+import {RasterDataTypes, VectorColumnDataType, VectorColumnDataTypes, VectorDataType, VectorDataTypes} from '../operators/datatype.model';
+import * as Immutable from 'immutable';
+import {Measurement} from './measurement';
+
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/layers/layer-metadata.ts:6]: again this is all ResultDescriptor IMHO

**Comment by michaelmattig** [comment]
Not sure where to put this but plots, like layers sould be renameable.

File: projects/wave-core-new/src/lib/layers/layer-metadata.ts (hunk 5 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/layers/layer-metadata.ts:6]: 53fbb49

File: projects/wave-core-new/src/lib/backend/backend.service.ts (hunk 4 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/backend/backend.service.ts]: 53fbb49

File: projects/wave-core-new/src/lib/backend/backend.model.ts (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/backend/backend.model.ts]: 53fbb49

File: projects/wave-core-new/src/lib/backend/backend.model.ts (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/backend/backend.model.ts]: 53fbb49

File: projects/wave-core-new/src/lib/backend/backend.model.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/backend/backend.model.ts]: 53fbb49

**Comment by ChristianBeilschmidt** [comment]
> Not sure where to put this but plots, like layers sould be renameable.

I don't quite understand what you mean but I guess renaming plots is a separate task.

**Comment by ChristianBeilschmidt** [comment]
> grafik
> 
>     * I don't see any labels yet, is that intended?
> 
>     * the bars are not aligned with the ticks


e2e4271 

There are labels and ticks now. The bar alignment is not 100% but this is up to vega.

**Comment by ChristianBeilschmidt** [comment]
> when going into full screen the bars don't get wider, even though there is space
> 
> grafik
> grafik

I guess vega tries to enforce some minimal padding which is too small for very large numbers of bars and small spaces. So for the large one, there is still not much space. I'm not quite sure what to do with it since it is auto layouted.

**Comment by michaelmattig** [comment]
Ok, I will merge it together with the backend PR

**Pull request merged**